Realm/Tests/AsyncTests.mm

```diff
@@ -721,11 +721,11 @@ - (void)testAddingBlockFromWithinNotificationBlock {
         }
     }];
 
-    // Triggers one call on each block
+    // Nested call is deferred.
     [self waitForNotification:RLMRealmDidChangeNotification realm:results.realm block:^{
         [self createObject:0];
     }];
-    XCTAssertEqual(calls, 2);
+    XCTAssertEqual(calls, 1);
```

Comment:
I think `XCTAssertEqual(calls, 1); [results.realm refresh]; XCTAssertEqual(calls, 2);` will work to test this behavior.

Reply:
Yes, that would better retain the original coverage of the test.